servers/src/main/java/tachyon/master/InodeFile.java

```diff
@@ -449,7 +449,10 @@ public synchronized void setLength(long length) throws SuspectedFileSizeExceptio
     if (length < 0) {
       throw new SuspectedFileSizeException("InodeFile new length " + length + " is illegal.");
     }
+
     mLength = 0;
+    mBlocks.clear();
```

Comment:
Logically we should always make sure that the blocks list of a file have been cleared before adding new blocks afterwards in `setLength`, functionally just like what `mLength = 0` does.